core/src/main/java/org/commonjava/maven/galley/io/SpecialPathManagerImpl.java

```diff
@@ -62,41 +62,27 @@ public void initPkgPathSets()
     }
 
     @Override
-    public void registerSpecialPathInfo( SpecialPathInfo pathInfo )
+    public synchronized void registerSpecialPathInfo( SpecialPathInfo pathInfo )
     {
-        //FIXME: do we need a flag of "isStandard" to add into stdSpecialPaths?
-        registerSpecialPathInfo( pathInfo, PKG_TYPE_MAVEN );
+        stdSpecialPaths.add( pathInfo );
     }
 
     @Override
     public void registerSpecialPathInfo( SpecialPathInfo pathInfo, final String pkgType )
     {
-        for ( Map.Entry<String, SpecialPathSet> entry : pkgtypes.entrySet() )
-        {
-            if ( pkgType.equals( entry.getKey() ) )
-            {
-                entry.getValue().registerSpecialPathInfo( pathInfo );
-            }
-        }
+        pkgtypes.get(pkgType).registerSpecialPathInfo( pathInfo );
     }
 
     @Override
     public synchronized void deregisterSpecialPathInfo( SpecialPathInfo pathInfo )
     {
-        //FIXME: do we need a flag of "isStandard" to remove from stdSpecialPaths?
-        deregisterSpecialPathInfo( pathInfo, PKG_TYPE_MAVEN );
+        stdSpecialPaths.remove( pathInfo );
     }
 
     @Override
     public void deregisterSpecialPathInfo( SpecialPathInfo pathInfo, String pkgType )
     {
-        for ( Map.Entry<String, SpecialPathSet> entry : pkgtypes.entrySet() )
-        {
-            if ( pkgType.equals( entry.getKey() ) )
-            {
-                entry.getValue().deregisterSpecialPathInfo( pathInfo );
-            }
-        }
+        pkgtypes.get(pkgType).deregisterSpecialPathInfo( pathInfo );
```

Comment:
Sometimes when I gives these types of code, it will lead me to a inertial thinking to loop and fetch firstly... or maybe some unknown mess in my mind makes me think in confusion.... :(